js/_reducers.js

```diff
@@ -0,0 +1,117 @@
+var _ = require('underscore');
+
+var actions = require('./_actions.js');
+
+// root reducer
+module.exports = function(state, action) {
+	if(typeof state === 'undefined') {
+		console.warn("state is undefined"); // eslint-disable-line no-console
+		return {};
+	}
+
+	switch(action.type) {
+
+		case actions.SET_MESSAGE:
+			console.log(action);
+			return _.assign({}, state, {
+				message: action.message,
+				message_type: action.message_type
+			});
+
+		case actions.SET_API_ENDPOINT:
+			return _.assign({}, state, {
```

Comment:
Sort of: I grabbed it of the redux docs http://redux.js.org/docs/basics/Reducers.html
```
Note on Object.assign

Object.assign() is a part of ES6, but is not implemented by most browsers yet. You’ll need to either use a polyfill, a Babel plugin, or a helper from another library like _.assign().
```
1) it uses the same assign syntax as ES6
2) it only copies the direct properties in case of .. extended props 

```
extendOwn _.extendOwn(destination, *sources) Alias: assign 
Like extend, but only copies own properties over to the destination object.
```